File: go/countries.go
package legalize

import (
	"context"
	"net/http"
)

// CountriesService exposes /api/v1/countries.
type CountriesService struct {
	client *Client
}

// List returns every country the API serves, with law counts.
func (s *CountriesService) List(ctx context.Context) ([]CountryInfo, error) {
	var out []CountryInfo
	if err := s.client.requestJSON(ctx, http.MethodGet, API+"/countries", nil, &out); err != nil {
		return nil, err
	}
	return out, nil
}

File: node/src/resources/countries.ts
/** `/api/v1/countries` — list every country the API serves. */

import type { Legalize } from "../client.js";
import type { CountryInfo } from "../types.js";

const API = "/api/v1";

export class Countries {
  private readonly client: Legalize;
  constructor(client: Legalize) {
    this.client = client;
  }

  /** Return every country the API serves, with law counts. */
  async list(options: { signal?: AbortSignal } = {}): Promise<CountryInfo[]> {
    return this.client.request<CountryInfo[]>("GET", `${API}/countries`, {
      ...(options.signal ? { signal: options.signal } : {}),
    });
  }
}
